integration-libs/cdc/root/service/cdc-js.service.ts

```diff
@@ -170,81 +170,112 @@ export class CdcJsService implements OnDestroy {
    *
-   * @param email
-   * @param password
+   * @param user: UserSignUp
    */
-  registerUserWithoutScreenSet(user: UserSignUp) {
-    if (user.uid && user.password) {
-      (this.winRef.nativeWindow as { [key: string]: any })?.[
-        'gigya'
-      ]?.accounts?.initRegistration({
-        callback: (response: any) => {
-          this.zone.run(() => this.onInitRegistrationHandler(user, response));
-        },
-      });
-    }
+  registerUserWithoutScreenSet(user: UserSignUp): Observable<Boolean> {
+    return new Observable<Boolean>((initRegistration) => {
+      if (user.uid && user.password) {
+        (this.winRef.nativeWindow as { [key: string]: any })?.[
+          'gigya'
+        ]?.accounts?.initRegistration({
+          callback: (response: any) => {
+            this.zone.run(() => {
+              this.onInitRegistrationHandler(user, response).subscribe(
+                (status) => {
+                  initRegistration.next(status);
+                  initRegistration.complete();
+                }
+              );
+            });
+          },
+        });
+      }
+    });
   }
 
   /**
    * Trigger CDC User registration using CDC APIs.
    *
    * @param response
    */
-  onInitRegistrationHandler(user: UserSignUp, response: any) {
-    if (response && response.regToken && user.uid && user.password) {
-      (this.winRef.nativeWindow as { [key: string]: any })?.[
-        'gigya'
-      ]?.accounts?.register({
-        email: user.uid,
-        password: user.password,
-        profile: {
-          firstName: user.firstName,
-          lastName: user.lastName,
-        },
-        regToken: response.regToken,
-        finalizeRegistration: true,
-        ignoreInterruptions: true,
-        callback: (response: any) => {
-          this.zone.run(() => {
-            if (response && response.status === 'FAIL') {
-              this.handleRegisterError(response);
-            }
-          });
-        },
-      });
-    }
+  protected onInitRegistrationHandler(
+    user: UserSignUp,
+    response: any
+  ): Observable<Boolean> {
+    return new Observable<Boolean>((isRegistered) => {
+      if (response && response.regToken && user.uid && user.password) {
+        (this.winRef.nativeWindow as { [key: string]: any })?.[
+          'gigya'
+        ]?.accounts?.register({
+          email: user.uid,
+          password: user.password,
+          profile: {
+            firstName: user.firstName,
+            lastName: user.lastName,
+          },
+          regToken: response.regToken,
+          finalizeRegistration: true,
+          callback: (response: any) => {
+            this.zone.run(() => {
+              if (response && response.status === 'FAIL') {
+                this.handleRegisterError(response);
+                isRegistered.next(false);
+              } else if (response && response.status === 'OK') {
+                isRegistered.next(true);
+              }
+              isRegistered.complete();
```

Comment:
The response from the CDC server is rich, but here we return only true/false. In the future we might want to return more information about the response, Eg. the error message.
What do you think about returning at lease an object, that contains (at least) the `status` property? Thanks to that, in the future we can add more properties to the response. TBH I wouldn't mind if we passed also the error message and other info that you think might be a good result of this function.

Similar for Login and Forgot Password.

Reply:
fixed in https://github.com/SAP/spartacus/pull/16160